manifests.go

```diff
@@ -0,0 +1,43 @@
+package distribution
+
+import "github.com/docker/distribution/digest"
+
+// Manifest specifies a registry object with a list of dependencies.
+type Manifest interface {
+	Blob // io access to serialized content.
+
+	// Dependencies returns a list of object on which this manifest depends.
+	// Typically, these are layers.
+	Dependencies() []Descriptor
+}
+
+// ManifestBuilder creates a manifest allowing one to include dependencies.
+// Instances can be obtained from a version-specific manifest package, likely
+// from a type that provides a version specific interface.
+type ManifestBuilder interface {
+	// Build creates the manifest fromt his builder.
+	Build() Manifest
+
+	// AddDependency includes the dependency in the manifest. If the add
+	// fails, such as when adding an unsupported dependency, an error may be
+	// returned.
+	AddDependency(dependency Descriptor) error
+}
+
+// ManifestService describes operations on image manifests.
```

Comment:
If I understand the purpose of this interface correctly, it is just to provide a nice wrapper around the BlobService when the media type is manifest. What other functionality would these calls add?  For example besides doing a blob write of the serialized manifest, what else does Put accomplish?